src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java

```diff
@@ -0,0 +1,578 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.AlwaysEncrypted;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.junit.jupiter.api.Test;
+
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerResultSet;
+import com.microsoft.sqlserver.testframework.util.Util;
+
+public class PrecisionScaleTest extends AESetup {
+    private static SQLServerPreparedStatement pstmt = null;
+
+    private static java.util.Date date = null;
+    private static int offsetFromGMT = 0;
+    private static int offset = 60000;
+    private static String GMTDate = "";
+    private static String GMTDateWithoutDate = "";
+    private static String dateTimeOffsetExpectedValue = "";
+
+    static {
+        TimeZone tz = TimeZone.getDefault();
+        offsetFromGMT = tz.getOffset(1450812362177L);
+
+        // since the Date object already accounts for timezone, subtracting the timezone difference will always give us the
+        // GMT version of the Date object. I can't make this PST because there are datetimeoffset tests, so I have to use GMT.
+        date = new Date(1450812362177L - offsetFromGMT);
+
+        // Cannot use date.toGMTString() here directly since the date object is used elsewhere for population of data.
+        GMTDate = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(date);
+        GMTDateWithoutDate = new SimpleDateFormat("HH:mm:ss").format(date);
+
+        // datetimeoffset is aware of timezone as well as Date, so we must apply the timezone value twice.
+        dateTimeOffsetExpectedValue = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+                .format(new Date(1450812362177L - offsetFromGMT - offsetFromGMT + offset));
+    }
+
+    @Test
```

Comment:
Great work. One last comment is to add javadocs for the test methods and a general description at the beginning of the file explaining what each class file is containing and testing.